frontend/src/components/pages/CreateReview/CreateReview.tsx

```diff
@@ -78,10 +101,35 @@ const CreateReview = (): React.ReactElement => {
     addBook(book);
   };
 
-  // useEffect hook adds dummy data to the books array
+  /**
+   * A callback function that maps a list of BookResponse objects
+   * to a list of Book objects and sets them in the component state
+   */
+  const setBooksFromBookResponse = useCallback(
+    (reviewResponse: ReviewResponse) => {
+      const result: Book[] = mapBookResponeToBook(reviewResponse.books);
+      setBooks(result);
+    },
+    [setBooks],
+  );
+
   useEffect(() => {
-    setBooks(data);
-  }, []);
+    if (id) {
+      reviewAPIClient
+        .getReviewById(id)
+        .then((reviewResponse: ReviewResponse) => {
+          setReview(reviewResponse.body);
+          setFeatured(reviewResponse.featured ? "1" : "0");
+          setReviewerByline(reviewResponse.byline);
+          setBooksFromBookResponse(reviewResponse);
+        })
+        .catch(() => {
+          history.push("/404");
```

Comment:
(minor) Is it possible to remove the current page from the history before pushing `/404`? Clicking the browser back button tries to fetch the invalid review again and re-navigates to the `404` page in an infinite loop